Modules/vc-module-pricing/VirtoCommerce.PricingModule.Web/Scripts/blades/assignment-list.js

```diff
@@ -189,14 +191,10 @@ angular.module('virtoCommerce.pricingModule')
                 }
             ];
 
-            var filter = $scope.filter = {};
             filter.criteriaChanged = function () {
-                if ($scope.pageSettings.currentPage > 1) {
-                    $scope.pageSettings.currentPage = 1;
-                } else {
-                    blade.refresh();
-                }
+                blade.refresh();
```

Comment:
Are you sure you need to remove 
```
                if ($scope.pageSettings.currentPage > 1) {
                    $scope.pageSettings.currentPage = 1;
                 }
```
We need to set page to 1 in case we are on another page. Refresh will be called by `pageSettings.currentPage`watch.